src/hermes/settings.py

```diff
@@ -0,0 +1,121 @@
+from pydantic import BaseModel
+import toml
+import pathlib
+from typing import Any, Dict, Tuple, Type, ClassVar
+
+from pydantic.fields import FieldInfo
+
+
+from pydantic_settings import (
+    BaseSettings,
+    PydanticBaseSettingsSource,
+    SettingsConfigDict,
+)
+
+
+class HarvestCff(BaseModel):
+    enable_validation: bool
+
+
+class HarvestSettings(BaseModel):
+    sources: list[str]
+    cff: HarvestCff = HarvestCff
+
+
+class DepositTargetSettings(BaseModel):
+    site_url: str
+    communities: list[str] = None
+    access_right: str
+
+
+class DepositSettings(BaseModel):
+    target: str
+    invenio: DepositTargetSettings = DepositTargetSettings
+    invenio_rdm: DepositTargetSettings = DepositTargetSettings
+
+
+class HermesSettings(BaseSettings):
+
+    model_config = SettingsConfigDict(env_file_encoding='utf-8', extra='allow')
+
+    harvest: HarvestSettings = HarvestSettings
+    deposit: DepositSettings = DepositSettings
+    config_path: ClassVar[pathlib.Path] = "hermes.toml" #TODO : Set config Path in cli
+
+    def __init__(self, conf_path):
+        super().__init__()
+        self.__class__.config_path = conf_path
+
+    @classmethod
+    def settings_customise_sources(
+            cls,
+            settings_cls: Type[BaseSettings],
+            init_settings: PydanticBaseSettingsSource,
+            env_settings: PydanticBaseSettingsSource,
+            dotenv_settings: PydanticBaseSettingsSource,
+            file_secret_settings: PydanticBaseSettingsSource,
+    ) -> Tuple[PydanticBaseSettingsSource, ...]:
+        return (
+            init_settings,
+            TomlConfigSettingsSource(settings_cls, cls.config_path),
+            env_settings,
+            file_secret_settings,
+        )
+    def __call__(self) -> Dict[str, Any]:
+        d: Dict[str, Any] = {}
+        for field_name, field in self.settings_cls.model_fields.items():
+            field_value, field_key, value_is_complex = self.get_field_value(
+                field, field_name
+            )
+            field_value = self.prepare_field_value(
+                field_name, field, field_value, value_is_complex
+            )
```

Comment:
These two statements could fit into one line if the maximum of 120 was asserted.